src/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import ListContainer from './ListContainer';
+import ListItemComponent from './ListItemComponent';
+
+const redditAPI = 'https://www.reddit.com/r/';
+
+export default class SubredditPosts extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoading: true,
+      redditPosts: []
+    };
+  }
+
+  componentDidMount() {
+    const uri = `${redditAPI}${this.props.subreddit}.json`;
+
+    fetch(uri)
+      .then(data => data.json())
+      .then(this.handlePosts)
+      .catch(err => console.error(err));
+  }
+
+  handlePosts = (posts) => {
+    const apiPosts = posts.data.children.map((post, index) => ({
+      key: index,
+      title: post.data.title,
+      media: this.getMediaFromPost(post),
+      link: post.data.url
+    }));
+
+    this.setState({
+      redditPosts: apiPosts,
+      isLoading: false
+    });
+  }
+
+  getMediaFromPost = (post) => {
+    const extension = post.data.url.split('.').pop();
+    if (post.data.hasOwnProperty('preiew') && extension === 'gif') {
+      return post.data.preview.preview.images[0].source.url;
+    }
+
+    if (extension === 'gif' || extension === 'jpg' || extension === 'jpeg') {
+      return post.data.url;
+    }
+
+    return this.props.placeholder;
+  }
+
+  render() {
+
+    return (
+      <ListContainer display={this.props.display}>
+        {this.state.isLoading && <p>Loading...</p>}
+        {!this.state.isLoading && this.state.redditPosts.map(post => (
+          <ListItemComponent
+            display={this.props.display}
+            key={post.key}
+            link={post.link}
+            media={post.media}
+            title={post.title}
+            height={this.props.height}
+            width={this.props.width}
```

Comment:
I'll dive into this a bit more in my next set of changes, but I wouldn't let your consumers control the width and height of your components. Your components should always respond to their parent's width to avoid any awkward breakpoints in responsive CSS.

Reply:
no i was wrong about how I wrote that. height and width actually control the size of each image. 
because originally when I wrote this, it was for imgur and in imgur photos are of different sizes and lengths. this created strange and ugly output.

Reply:
Take a look at possibly using `object-fit` in CSS to determine a certain box and force the image to fit into it without compromising the ratio: https://css-tricks.com/almanac/properties/o/object-fit/

Another way is to use CSS Grid and do a Masonry look: https://medium.com/@andybarefoot/a-masonry-style-layout-using-css-grid-8c663d355ebb

Reply:
these are interesting. I'll consider this for the imgur react component. 

the grid layout is actually designed for a friend's website and it's exactly how he wants it. 
So I'm not going to try and touch it up too much unfortunately but that's an amazing design solution for the different sizes problem.